File: Lab 2/1002141462_calculator.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int isValid(const char *str){
    int i= 0;
    
}

int main(){
    double x,y, total;
    char symbol, inputa[25], inputb[25];
    Printf("Welcome to the calculator. \n");
    printf("Enter the first number: ");
    scanf("%lf" &x);
    printf("Enter the symbol (+, -, *,/): ): ");
    scanf("%c" &symbol);
    printf("Enter the second number: ");
    scanf("%lf" &y);

    if(symbol==+){
        return a + y;
    }else if(symbol == -){
        return a + y;
    }else if(symbol == *){
        return a * y;
    }else if(symbol == /){
        if(b==0){
            printf("Unvalid");
            return 1;
        }else{
            return a / y;
        }

    }else{
        printf("invalid calculation");
        return 1;
    }
    printf("Result: %.2f\n", total);
    return 0;



}

File: Lab 2/1002141462_calculator.py
#functions have all of the operations with x and y as input

def additon(x,y):
    return x+y
def subtraction(x,y):
    return x-y
def multiplication(x,y):
    return x*y
def division(x,y):
    if y==0: #does a check for division by 0
        return "Invalid computation"
    return x/y

def values(inputs):
    while True:
        try:
            return float(input(inputs))
        except ValueError:
            print("Invalid number, please enter a new number")

def operation():
    # will promt a user for a valid input until they give one
    while True:
        symbol = input("Enter the symbol (+, -, *,/): ")
        if symbol in('+', '-','*','/'):
            return symbol
        print("Invalid number, please enter a new symbol (+, -, *,/).")

def calculator (x, symbol, y):#helps map the function to the symbol
    operation={'+': additon, 
               '-': subtraction,
               '*': multiplication,
               '/': division}
    return operation[symbol](x,y)

# prompts the user and gets input
x = values("Enter your first number")
symbol = operation()
y = values("Enter your second number")
print("The result is :", calculator(x, symbol, y))